test/AllowListStorageReader.test.ts

```diff
@@ -0,0 +1,84 @@
+import { expect } from "chai";
+import { BigNumber, Contract } from "ethers";
+import { ethers, waffle } from "hardhat";
+
+const readStorage = async (
```

Comment:
I would expect this to not be test code but exported under `index/ts/reader.ts` or the like as I expect others to want to be able to use this code as well.

It would also be nice to have a class that wraps these methods together to make it easier to use:
```ts
export class AllowListStorageReader {
  constructor(public readonly allowList:  Contract, public readonly reader: Contract) {}
  public getSolverAt(index: number): Promise<string> {...}
  public numSolvers(): Promise<number> {...}
}
```